### PR TITLE
Basic integration test

### PR DESCRIPTION
**Pull request by alokmenghrajani**
I still need to improve some things (make freshTimeout a parameter, remove the hardcoded alok/staff, etc.). Putting this here to get some initial feedback.

### CODE REVIEW

**Comment by csstaub** [comment]
Very cool.

**Comment by csstaub** [comment]
Don't forget to update the makefile so that the new tests also run on travis-ci.

**Comment by csstaub** [comment]
@alokmenghrajani I made it such that the integration tests also run in travis-ci now. Caveat is that the integration tests only run on Linux for now, will break on OS X because the `fusermount` command doesn't exist. We can fix that later with some conditionals if we want.

File: cache_test.go [hunk 1]
```diff
@@ -210,6 +210,31 @@ func TestCacheSecretAvoidsBackendWhenResultFresh(t *testing.T) {
 	assert.Equal(fixture1, secret) // fixture1 comes form the backend
 }
 
+func TestCacheSecretUsesBackendWhenResultStale(t *testing.T) {
```
Comment (alokmenghrajani) [inline cache_test.go]: this test is not useful because it doesn't repro our issue.

File: integration-tests/integration_test.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// start keywhiz-fs
+	os.MkdirAll("./mount", 0755)
+	kwfs = exec.Command("../keywhiz-fs", "--cert", "kwfs.crt", "--key",
+		"kwfs.key", "--ca", "keywhiz.crt", "--debug", "--disable-mlock",
+		"--asuser", user.Username, "--group", lookupGroup(user.Gid),
+		"--cache-timeout", "1s", "https://localhost:8080/", "./mount")
```
Comment (alokmenghrajani) [inline integration-tests/integration_test.go]: We could have the test spin up the fake server on a random port and then connect to it here. This way we don't need to build a binary and we won't hardcode a port.

**Comment by alokmenghrajani** [comment]
let's merge this and improve on it?

File: cache_test.go (hunk 1 above)
Comment (csstaub) [inline cache_test.go]: It fails on macOS though so it certainly repros the issue with osxfuse we have.

**Pull request merged**